### DISCUSSION TITLE
tart images pushed to gitlab container registry always show a publish date of right now

### DISCUSSION THREAD

**Discussion by cosmoshepcat** [cirruslabs/tart#603] [Q&A] [answered]
Curious if anyone else has run into this but since these are not really docker images its probably not surprising.  

if i do a tart push of a local vm to our remote container registry it works fine but in the gitlab ui it shows every image as always freshly pushed.  refreshing over and over shows the image publish date updates with the time.   i am assuming this is because some docker info is missing in a tart image that gitlab is looking for.

not a show stopper but this does make auto cleanup on various tart repos within gitlab impossible since it uses those publish dates.

<img width="1360" alt="Screenshot 2023-09-20 at 4 43 14 PM" src="https://github.com/cirruslabs/tart/assets/22826883/3f472732-d082-4349-92f5-0d72ca06e7d1">

**Comment by edigaryev** [comment, marked as answer]
This looks like a purely GitLab Registry's issue to me.

If you toggle the `Use relative times` setting in `Preferences → Time preferences`, you'll see that GitLab Registry simply leaves the timestamp uninitialized for manifests/configs that they fail to recognize and pull the timestamp from<sup>1</sup>:

<img width="278" alt="Screenshot 2023-09-21 at 10 57 34" src="https://github.com/cirruslabs/tart/assets/85709/a788b4f6-8bf1-49f4-840e-1c8fa928f833">

Compare this to GitHub:

<img width="745" alt="Screenshot 2023-09-21 at 14 59 57" src="https://github.com/cirruslabs/tart/assets/85709/1b505704-dc64-4f3e-a66d-49ae596b007c">

<sup>1</sup>: GitLab Registry pulls the `created` field from the manifest configuration object of type `application/vnd.docker.container.image.v1+json`. However, Tart uses a different media type of `application/vnd.oci.image.config.v1+json`. There exist a well-known annotation `oci.opencontainers.image.created` [mentioned in the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage), but GitLab Registry doesn't seem to support it, judging from [their source code](https://gitlab.com/gitlab-org/container-registry).
